sklearn/tree/_criterion.pyx

```diff
@@ -962,6 +969,304 @@ cdef class MSE(RegressionCriterion):
         impurity_left[0] /= self.n_outputs
         impurity_right[0] /= self.n_outputs
 
+cdef class MAE(RegressionCriterion):
```

Comment:
One other important thing, predictions at leaves should be those that minimize the chosen criterion. This is why this class should also override `node_value` in order to predict the median at leaves, and not the mean as it is by default in `RegressionCriterion`.